app/HutchAgent/Services/CrateService.cs

```diff
@@ -333,4 +333,40 @@ public void DeleteContainerImages(string pathToImagesDir)
   {
     Directory.Delete(pathToImagesDir, recursive: true);
   }
+
+  public void CheckAssessActions(ROCrate roCrate)
+  {
+    //Check CheckValueType AssessAction exists and is Completed
+    var checkValue = GetAssessAction(roCrate, ActionType.CheckValueType);
+    if (checkValue is null)
+      throw new Exception("Could not find CheckValue AssessAction in RO-Crate");
+
+    var checkValueStatus = checkValue.GetProperty<JsonNode>("actionStatus");
+    if (checkValueStatus?.ToString() is not (ActionStatus.CompletedActionStatus))
```

Comment:
would this `?` cause an unexpected exception different to the one you throw below?

Reply:
the `?` is fine because it'll result in `null is not ActionStatus.CompletedActionStatus` which will safely return true and throw the exception as expected.

However, I would use `!=` rather than `is not` here?